TypedSyntax/src/vscode.jl

```diff
@@ -0,0 +1,80 @@
+isvscode() = isdefined(Main, :VSCodeServer) && Main.VSCodeServer isa Module
+inlay_hints_available() = isvscode() && isdefined(Main.VSCodeServer, :INLAY_HINTS_ENABLED)
+
+function clearall()
+    if inlay_hints_available()
+        display(Main.VSCodeServer.InlineDisplay(false),  Dict{String, Vector{TypedSyntax.InlayHint}}())
+    end
+    if isvscode()
+        display(Main.VSCodeServer.InlineDisplay(false), TypedSyntax.WarnUnstable[])
+    end
+end
+
+struct WarnUnstable
+    path::String
+    line::Int
+    severity::Int # 0: Error, 1: Warning, 2: Information, 3: Hint
+end
+to_vscode_type(x::WarnUnstable) = (msg="Unstable Type", path = x.path, line = x.line, severity = x.severity)
+function Base.show(io::IO, ::MIME"application/vnd.julia-vscode.diagnostics", warn_diagnostics::AbstractVector{WarnUnstable})
+    return (
+        source = "Cthulhu",
+        items = to_vscode_type.(warn_diagnostics),
+    )
+end
+function add_diagnostic!(warn_diagnostics, node, position, severity)
+    file_path = node.filename
+    line, column = source_location(node, position)
+    push!(warn_diagnostics, WarnUnstable(file_path, line, severity))
+end
+
+const InlayHintKinds = (
```

Comment:
Should these be an `@enum`?

Reply:
As InlayHintKinds can also be `nothing` I've left it as a constant. I changed the severity in WarnUnstable (now Diagnostic) to an @enum, I put it in a module to avoid polluting the global namespace but I can change that.